components/custom_open_plugin_install_post_renderer/index.tsx

```diff
@@ -0,0 +1,335 @@
+
+import React, {useEffect, useState} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+import {useSelector, useDispatch} from 'react-redux';
+
+import {Link} from 'react-router-dom';
+
+import {uniqWith} from 'lodash';
+
+import MarketplaceModal from 'components/plugin_marketplace';
+
+import Markdown from 'components/markdown';
+
+import {getUsers} from 'mattermost-redux/selectors/entities/users';
+
+import {ModalIdentifiers} from 'utils/constants';
+
+import {getMissingProfilesByIds} from 'mattermost-redux/actions/users';
+
+import {localizeMessage} from 'utils/utils';
+
+import {Post} from '@mattermost/types/posts';
+import {MarketplacePlugin} from '@mattermost/types/marketplace';
+import {GlobalState} from '../../types/store';
+import {installPlugin} from '../../actions/marketplace';
+import ToggleModalButton from '../toggle_modal_button';
+import {getError, getInstalledListing, getInstalling} from '../../selectors/views/marketplace';
+import LoadingWrapper from '../widgets/loading/loading_wrapper';
+
+type PluginRequest = {
+    user_id: string;
+    required_feature: string;
+    required_plan: string;
+    create_at: string;
+    sent_at: string;
+    plugin_name: string;
+    plugin_id: string;
+}
+
+type RequestedPlugins = Record<string, PluginRequest[]>
+
+type CustomPostProps = {
+    requested_plugins_by_plugin_ids: RequestedPlugins;
+    requested_plugins_by_user_ids: RequestedPlugins;
+}
+
+const buttonsStyle = {
+    display: 'flex',
+    gap: '10px',
+    padding: '12px',
+    borderRadius: '4px',
+    border: '1px solid rgba(var(--center-channel-color-rgb), 0.16)',
+    width: 'max-content',
+    margin: '10px 0',
+};
+
+const usersListStyle = {
+    margin: '20px 0',
+};
+
+export default function OpenPluginInstallPost(props: {post: Post}) {
+    const customMessageBody = [];
+
+    const dispatch = useDispatch();
+    const {formatMessage, formatList} = useIntl();
+    const [pluginsByPluginIds, setPluginsByPluginIds] = useState<RequestedPlugins>({});
+
+    const postProps = props.post.props as CustomPostProps;
+    const requestedPluginsByPluginIds = postProps?.requested_plugins_by_plugin_ids;
+    const requestedPluginsByUserIds = postProps?.requested_plugins_by_user_ids;
+
+    const userProfiles = useSelector(getUsers);
+    const marketplacePlugins: MarketplacePlugin[] = useSelector((state: GlobalState) => state.views.marketplace.plugins);
+
+    const getUserIdsForUsersThatRequestedFeature = (requests: PluginRequest[]): string[] => requests.map((request: PluginRequest) => request.user_id);
+
+    useEffect(() => {
+        if (requestedPluginsByPluginIds && marketplacePlugins.length && !Object.keys(pluginsByPluginIds).length) {
+            let plugins = {} as RequestedPlugins;
+            for (const pluginId of Object.keys(requestedPluginsByPluginIds)) {
+                const plugin = marketplacePlugins.find((marketplacePlugin) => marketplacePlugin.manifest.id === pluginId);
```

Comment:
nit: making a dictionary just above the loop would prevent having to do the find at every iteration